src/arviz_plots/plots/violin_plot.py

```diff
@@ -0,0 +1,331 @@
+"""violin plot code."""
+
+import warnings
+from collections.abc import Mapping, Sequence
+from copy import copy
+from importlib import import_module
+from typing import Any, Literal
+
+import arviz_stats  # pylint: disable=unused-import
+import xarray as xr
+from arviz_base import rcParams
+from arviz_base.labels import BaseLabeller
+
+from arviz_plots.plot_collection import PlotCollection
+from arviz_plots.plots.utils import filter_aes, process_group_variables_coords, set_wrap_layout
+from arviz_plots.visuals import line_xy, line_y
+
+
+def plot_violin(
+    dt,
+    var_names=None,
+    filter_vars=None,
+    group="posterior",
+    coords=None,
+    sample_dims=None,
+    point_estimate=None,
+    ci_kind=None,
+    ci_prob=None,
+    plot_collection=None,
+    backend=None,
+    labeller=None,
+    aes_by_visuals: Mapping[
+        Literal[
+            "dist",
+            "credible_interval",
+            "point_estimate",
+            "point_estimate_text",
+            "title",
+            "rug",
+        ],
+        Sequence[str],
+    ] = None,
+    visuals: Mapping[
+        Literal[
+            "dist",
+            "credible_interval",
+            "point_estimate",
+            "point_estimate_text",
```

Comment:
I don't think we'll want the point estimate text in this plot, the other ones can be nice, if you haven't added it yet I don't think it is necessary you do at all